MultiPoolMiner.ps1

```diff
@@ -712,7 +712,7 @@ while (-not $API.Stop) {
     $API.WatchdogTimers = $WatchdogTimers #Give API access to WatchdogTimers information
     if ($Downloader) {$Downloader | Receive-Job}
     Start-Sleep $Config.Delay #Wait to prevent BSOD
-    Get-CIMInstance CIM_Process | Where-Object ExecutablePath | Where-Object {$_.ExecutablePath -like "$(Get-Location)\Bin\*"} | Where-Object {$ActiveMiners.ProcessID -notcontains $_.ProcessID} | Select-Object -ExpandProperty ProcessID | ForEach-Object {Stop-Process -Id $_ -Force -ErrorAction Ignore} #Kill stray miners
+    if ($ActiveMiners | ForEach-Object {$_.GetProcessNames()}) {Get-Process -Name @($ActiveMiners | ForEach-Object {$_.GetProcessNames()} | Select-Object) -ErrorAction Ignore | Select-Object -ExpandProperty ProcessName | Compare-Object @($ActiveMiners | Where-Object Best | Where-Object {$_.GetStatus() -eq "Running"} | ForEach-Object {$_.GetProcessNames()} | Select-Object) | Where-Object SideIndicator -EQ "=>" | Select-Object -ExpandProperty InputObject | Select-Object -Unique | ForEach-Object {Stop-Process -Name $_ -Force -ErrorAction Ignore}} #Kill stray miners
```

Comment:
This does NOT kill any miners that might still be running from previous runs, or when MPM run stated twice from the same directory.
I don't understand why you are so opposed to my solution.

Reply:
The 'bin' folder is not a folder used by MultiPoolMiner. There are two lists being compared; one from the miners and one from MultiPoolMiner. Sounds like there is probably a bug in one of the lists.

Reply:
>> The 'bin' folder is not a folder used by MultiPoolMiner.

Yes it is:
$_.ExecutablePath -like "$(Get-Location)\Bin\*" ponts to the MPM\Bin folder.

Reply:
The 'Bin' folder is definitely nowhere within `MultiPoolMiner.ps1` or `Include.psm1`. 

I'll have a look at this soon though as something must be wrong here.

Reply:
@UselessGuru I think what @aaronsace is *trying* to say is that the binaries are currently downloaded to .\Bin, but they don't *have* to be, and he maybe intents to change that in the future.  Please correct me if I'm wrong there.

> A `Miner` has control over everything including what files should/shouldn't be running. The bin folder is abritory; a miner could be running files from Program Files, a network share or anywhere. 

The problem is, the Miner object *should* reflect what miners, if any, are currently running, but reality doesn't always match what the objects say.  There can be miners still running from:

- a previous run of MPM that crashed or otherwise didn't clean up when it exited
- making changes to the miner files without restarting MPM
- a miner being stopped on the last cycle, but windows not cooperating, saying it was killed, and it still running.  Windows refusing to actually close a program when you tell it to isn't a code problem, it's a windows problem.  Everyone has at some point had a window where they could click the X all they wanted and nothing happens.  Sometimes even trying to kill apps from task manager fails.  It's just a windows thing.  Some of the miners take awhile to actually exit, and don't actually finish exiting on the first try, especially on systems with slow CPUs.


So the solution we put in way back when was to look at the Miner objects, look at the running processes, and try to kill any executable running in the .\Bin folder that shouldn't be running according to the Miner objects.

This brings reality back into line with what the Miner objects think is happening.

Limiting to killing executables in the .\Bin folder was to prevent it from interfering with any other executables - say if you had other miners installed elsewhere you ran along side MPM.

One possible alternative that doesn't rely on them being in the .\Bin folder would be to build a list of all the full paths of the miner executables, comparing THAT list to the "expected to be running" miners, and killing any that shouldn't be running.  Changing the miner definition files without restarting MPM could still leave something running, and it would probably be a little slower than just brute force killing whatever is in the .\Bin folder, but it would solve the other two ways I know of miners being accidentally left running.

Reply:
> One possible alternative that doesn't rely on them being in the .\Bin folder would be to build a list of all the full paths of the miner executables,

To fetch ALL miner binary paths a separate Get-ChildItemContent with all HW activated would be needed, otherwise we might build an incomplete list that could be missing miners that have been disabled since the last run.

I'd stick with the Bin\ path version as long as all miners are using that path.